### PR TITLE
Move classes declared in GenericConsole all to their own files

### PR DESCRIPTION
**Pull request by DanVanAtta**
Just a straight move

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/928)

### CODE REVIEW

File: src/games/strategy/debug/ThreadReader.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class ThreadReader implements Runnable {
+  private static final int CONSOLE_UPDATE_INTERVAL_MS = 100;
+  private final JTextArea m_text;
+  private final SynchedByteArrayOutputStream m_in;
+  private final boolean m_displayConsoleOnWrite;
+  private final GenericConsole parentConsole;
+
+  ThreadReader(final SynchedByteArrayOutputStream in, final JTextArea text, final boolean displayConsoleOnWrite,
+      GenericConsole parentConsole) {
+    m_in = in;
```
Comment (RoiEXLab) [inline src/games/strategy/debug/ThreadReader.java]: Variable prefix could be removed

**Comment by RoiEXLab** [comment]
@DanVanAtta I like auxilliary classes being moved in their own source files, but they should made public then...
Also, when moving then you could remove their variable prefixes.
In all of your PRs

**Comment by DanVanAtta** [comment]
For this type of change, because github is showing the code as removed and added, I want to minimize the actual differences. If things start changing, then we will need to run diff's by hand. Secondly, the scope of this PR is pretty straight forward. Ideally we would not spend huge amounts of time studying code for every improvement that could be made, whether it was changed or not, but  instead verifying that changes are reasonable and accomplish what is intended.

**Comment by RoiEXLab** [comment]
@DanVanAtta No objections then...

**Pull request merged**